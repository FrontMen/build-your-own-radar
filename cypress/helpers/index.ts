export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
