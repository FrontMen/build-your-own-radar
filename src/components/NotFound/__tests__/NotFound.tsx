import React from 'react';
import { NotFound } from '..';
import { withAllProviders } from 'test/helpers';

describe('NotFound', () => {
  test('matches the snapshot', () => {
    const { container, getByTestId } = withAllProviders(<NotFound />);

    expect(container).toMatchSnapshot();
    expect(getByTestId('not-found-content')).toHaveTextContent('Page not found');
  });
});
