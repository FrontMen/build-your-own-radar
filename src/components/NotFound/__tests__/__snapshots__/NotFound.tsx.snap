// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound matches the snapshot 1`] = `
.c0 {
  grid-area: main;
  background-color: #fff;
  padding: 16px;
}

.c4 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 0.9em 1.25em;
  padding-right: 3.5em;
  color: #fff;
  background-color: #526de4;
  border-color: #526de4;
}

.c4 img {
  position: absolute;
  top: 15px;
  right: 15px;
  width: 1.4em;
  height: 1.1em;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.23,1,0.32,1);
  -webkit-transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
  transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
}

.c4:hover img {
  -webkit-transform: translateX(5px);
  -ms-transform: translateX(5px);
  transform: translateX(5px);
}

.c1 {
  font-family: Montserrat,san-serif;
  font-weight: 700;
  line-height: 1.15;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 50vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 140px;
  font-weight: 200;
  margin: 0;
  background: linear-gradient(130deg,#ffa34f,#ff6f68);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  text-transform: uppercase;
  text-align: center;
}

.c3 {
  font-size: 25px;
  font-weight: 200;
  text-transform: uppercase;
  margin-top: 0px;
  margin-bottom: 50px;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

@media (min-width:481px) {
  .c0 {
    padding: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="not-found-content"
    >
      <div>
        <h1
          class="c2"
        >
          404
        </h1>
        <h2
          class="c3"
        >
          <span>
            notfound.description
          </span>
        </h2>
        <a
          class="c4"
          data-testid="not-found-link"
          href="/"
        >
          <span>
            notfound.link
          </span>
          <img
            alt="right-arrow"
            src="arrow-right.svg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
